### PR TITLE
feat(storage): add grpc metrics experimental options

### PR DESCRIPTION
**Pull request by frankyn**

### CODE REVIEW

File: storage/experimental/experimental.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+import (
+	"time"
+
+	"cloud.google.com/go/storage/internal"
+	"go.opentelemetry.io/otel/sdk/metric"
+	"google.golang.org/api/option"
+)
+
+// WithMetricInterval how often to emit metrics when using NewPeriodicReader
```
Comment (tritone) [inline storage/experimental/experimental.go:27]: Make it clear that these options are supposed to be used with storage.NewClient.

**Review by tritone — COMMENTED** [review]
Looks great overall, thanks for figuring this out Frank!

**Comment by tritone** [comment]
Also, looks like your unit test fails... can you investigate?

File: storage/experimental/experimental.go [hunk 2]
```diff
@@ -25,9 +25,27 @@ import (
 	"time"
 
 	"cloud.google.com/go/storage/internal"
+	"go.opentelemetry.io/otel/sdk/metric"
 	"google.golang.org/api/option"
 )
 
+// WithMetricInterval provides a [ClientOption] that may be passed to [storage.NewGrpcClient].
+// It sets how often to emit metrics when using NewPeriodicReader and must be
+// greater than 1 minute.
```
Comment (tritone) [inline storage/experimental/experimental.go:34]: Why is this the minimum? Not sure why this requires validation, users might have their own systems with higher rate limits.

File: storage/experimental/experimental.go [hunk 3]
```diff
@@ -25,9 +25,27 @@ import (
 	"time"
 
 	"cloud.google.com/go/storage/internal"
+	"go.opentelemetry.io/otel/sdk/metric"
 	"google.golang.org/api/option"
 )
 
+// WithMetricInterval provides a [ClientOption] that may be passed to [storage.NewGrpcClient].
```
Comment (tritone) [inline storage/experimental/experimental.go:32]: typo: should be storage.NewGRPCClient (here and below)

File: storage/experimental/experimental.go [hunk 4]
```diff
@@ -25,9 +25,27 @@ import (
...
+// It sets how often to emit metrics when using NewPeriodicReader and must be
+// greater than 1 minute.
+// https://pkg.go.dev/go.opentelemetry.io/otel/sdk/metric#NewPeriodicReader
```
Comment (tritone) [inline storage/experimental/experimental.go:35]: leave an empty line (with a `//` only) between links if you want them to not wrap in the godoc.

File: storage/experimental/experimental.go [hunk 5]
```diff
@@ -25,9 +25,27 @@ import (
...
+// https://pkg.go.dev/go.opentelemetry.io/otel/sdk/metric#WithInterval
+func WithMetricInterval(metricInterval time.Duration) option.ClientOption {
+	return internal.WithMetricInterval.(func(time.Duration) option.ClientOption)(metricInterval)
+}
+
+// WithMetricExporter provides a [ClientOption] that may be passed to [storage.NewGrpcClient].
+// Set an alternate client-side metric Exporter to emit metrics through.
+// Must implement interface metric.Exporter:
```
Comment (tritone) [inline storage/experimental/experimental.go:43]: Just do [metric.Exporter] and the godoc should autolink.

File: storage/option.go [hunk 6]
```diff
@@ -15,12 +15,14 @@
 package storage
 
 import (
+	"time"
```
Comment (tritone) [inline storage/option.go:18]: Not sure why this got moved but it will probably break formatting presubmits

**Review by tritone — COMMENTED** [review]
Couple minor nits/questions, overall looks good

File: storage/experimental/experimental.go (hunk 3 above)
Comment (BrennaEpp) [inline storage/experimental/experimental.go:32]: nit: what does ClientOption link to?

File: storage/experimental/experimental.go (hunk 2 above)
Comment (frankyn) [reply storage/experimental/experimental.go:34]: Good point; could pass along a value if it's non-zero.

File: storage/experimental/experimental.go [hunk 7]
```diff
@@ -25,10 +25,25 @@ import (
...
 
 	"cloud.google.com/go/storage/internal"
+	"go.opentelemetry.io/otel/sdk/metric"
 	"google.golang.org/api/option"
 )
 
-// WithReadStallTimeout provides a [ClientOption] that may be passed to [storage.NewClient].
+// WithMetricInterval provides a [option.ClientOption] that may be passed to [storage.NewGRPCClient].
+// It sets how often to emit metrics [metric.WithInterval] when using
+// [metric.NewPeriodicReader]
```
Comment (tritone) [inline storage/experimental/experimental.go:34]: I think removing the minimum setting is good. Maybe suggest time.Minute or higher as an interval in the godoc if exporting to cloud monitoring.

**Review by tritone — APPROVED** [review]
One more doc comment, otherwise LGTM

**Review by Jenniferab5 — COMMENTED** [review]
Review

**Pull request merged**